packages/runner/injection/patches/cookies.js

```diff
@@ -0,0 +1,130 @@
+import { Cookie } from 'tough-cookie'
+
+// document.cookie monkey-patching
+// -------------------------------
+// We monkey-patch document.cookie when in a cross-origin injection, because
+// document.cookie runs into cross-origin restrictions when the AUT is on
+// a different origin than top. The goal is to make it act like it would
+// if the user's app was run in top.
+//
+// The general strategy is:
+// - Keep the document.cookie value (`documentCookieValue`) available so
+//   the document.cookie getter can synchronously return it.
+// - Optimistically update that value when document.cookie is set, so that
+//   subsequent synchronous calls to get the value will work.
+// - On an interval, get the browser's cookies for the given domain, so that
+//   updates to the cookie jar (via http requests, cy.setCookie, etc) are
+//   reflected in the document.cookie value.
+export const patchDocumentCookie = (win) => {
+  const getCookiesFromCypress = () => {
+    return new Promise((resolve, reject) => {
+      const handler = (event) => {
+        if (event.data.event === 'cross:origin:aut:get:cookie') {
```

Comment:
so this is mostly for the case of something like 

```
cy.visit('cross-origin')
cy.getCookie('foo')
cy.origin...
```

The injection has a second to respond with fetched automation cookies to the AUT otherwise the command fails?

Reply:
This funciton is used to sync cookies to document.cookie on the cross-origin AUT if cookie changes are made prior to loading the spec bridge but after load. I think this was to handle the case where we use CDP to set cookies after load (this would be done in the primary). I think this will change with chris' changes.